src/Cli/dotnet/commands/dotnet-workload/install/WorkloadManifestUpdater.cs

```diff
@@ -45,9 +51,52 @@ internal class WorkloadManifestUpdater : IWorkloadManifestUpdater
         public async Task UpdateAdvertisingManifestsAsync(bool includePreviews, DirectoryPath? offlineCache = null)
         {
             var manifests = GetInstalledManifestIds();
-            foreach (var manifest in manifests)
+            await Task.WhenAll(manifests.Select(manifest => UpdateAdvertisingManifestAsync(manifest, includePreviews, offlineCache)))
+                .ConfigureAwait(false);
+        }
+
+        public async static Task BackgroundUpdateAdvertisingManifestsStaticAsync()
+        {
+            try
+            {
+                var reporter = new NullReporter();
+                var dotnetPath = Path.GetDirectoryName(Environment.ProcessPath);
+                var sdkVersion = Product.Version;
+                var workloadManifestProvider = new SdkDirectoryWorkloadManifestProvider(dotnetPath, sdkVersion);
+                var workloadResolver = WorkloadResolver.Create(workloadManifestProvider, dotnetPath, sdkVersion);
+                var tempPackagesDir = new DirectoryPath(Path.Combine(Path.GetTempPath(), "dotnet-sdk-advertising-temp"));
+                var nugetPackageDownloader = new NuGetPackageDownloader(tempPackagesDir,
+                                              filePermissionSetter: null,
+                                              new FirstPartyNuGetPackageSigningVerifier(tempPackagesDir, new NullLogger()),
+                                              new NullLogger(),
+                                              reporter);
+                var userHome = CliFolderPathCalculator.DotnetHomePath;
+
+                var manifestUpdater = new WorkloadManifestUpdater(reporter, workloadManifestProvider, workloadResolver, nugetPackageDownloader, userHome, tempPackagesDir.Value);
+                await manifestUpdater.BackgroundUpdateAdvertisingManifestsAsync();
+            }
+            catch (Exception)
+            {
+                // Never surface messages on background updates
+            }
+}
+
+        public async Task BackgroundUpdateAdvertisingManifestsAsync()
```

Comment:
nit: call them BackgroundUpdateAdvertisingManifestsWhenRequiredAsync and BackgroundUpdateAdvertisingManifestsAsync ?

It is not obvious to read BackgroundUpdateAdvertisingManifestsStaticAsync vs BackgroundUpdateAdvertisingManifestsAsync to understand the difference